server/src/main/java/org/elasticsearch/index/mapper/StoredValueFetcher.java

```diff
@@ -38,7 +38,12 @@ public void setNextReader(LeafReaderContext context) {
     @Override
     public List<Object> fetchValues(Source source, int doc, List<Object> ignoredValues) throws IOException {
         leafSearchLookup.setDocument(doc);
-        return leafSearchLookup.fields().get(fieldname).getValues();
+        List<Object> values = leafSearchLookup.fields().get(fieldname).getValues();
+        return List.copyOf(values);
+            // copy _id field to avoid field value cache clear
+            values = List.copyOf(values);
+        }
+        return values;
```

Comment:
This broke a bit after accepting the last suggestion, sorry for that. Can you also add a null check to the values passed to List.copyOf(), otherwise I think e.g. in case of the ignore field this can potentially throw NPEs

Reply:
Yes, I am checking.

Reply:
@cbuescher Would you please help to check again? Thanks.